src/auryn/StateMonitor.h

```diff
@@ -130,6 +139,28 @@ class StateMonitor : public Monitor
 	void execute();
 };
 
+/*! \brief Records from an arbitray pretrace vector of one unit from the source SpikingGroup to a file.*/
+class PreTraceMonitor : public StateMonitor
```

Comment:
What do we need those for? StateMonitor can take an arbitrary Trace as argument and record from it. It seems like this Class isn't doing anything. If this is for readability I would suggest putting it as a typedef into the simulation code where it's being used.